### PR TITLE
[rfc] bash completion improvements

### PR DESCRIPTION
**Pull request by thomas-gardner**
This completion script is a little smarter than the current one. Some selling points are that it:

 * generates matches on the fly
 * supports file/peer/pin completion
 * is probably easier to hack on

Let me know if it works.

### CODE REVIEW

**Comment by thomas-gardner** [comment]
N.B., Github wraps commit messages before 75 colzzz

File: misc/completion/ipfs-completion.bash [hunk 1]
```diff
@@ -1,421 +1,84 @@
...
+	if [[ -z "$reply" ]]; then
+		local prev="${COMP_WORDS[COMP_CWORD-1]}"
+		case "$prev" in
+			# these are pretty crude
+			id|ping)
+				reply="$(__ipfs_peers_list)"
+				compopt -o nospace
+				;;
+			cat|recursive|direct|indirect|all)
+				reply="$(__ipfs_pin_list)"
```
Comment (whyrusleeping) [inline misc/completion/ipfs-completion.bash]: this is going to be painfully slow.

**Comment by whyrusleeping** [comment]
I think @chriscool and @lgierth might have some opinions here

File: misc/completion/ipfs-completion.bash (hunk 1 above)
Comment (thomas-gardner) [inline misc/completion/ipfs-completion.bash]: > this is going to be painfully slow.

You're talking about the $(ipfs refs local)?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: dc80a3dee519a4f0ab5920a8a0e268ec9726d7a5
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

We ask for a few features in the commit message for Open Source licensing hygiene and commit message clarity.
`git commit --amend` can often help you quickly improve the commit message.
[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) and [a script](https://github.com/ipfs/community/blob/master/dev/tools/hooks/setup_commit_msg_hook.sh) are available to help in the long run.
Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23).

---
This message was auto-generated by https://gitcop.com

**Comment by RichardLitt** [comment]
Can we kill GitCop? @chriscool what is happening here.

**Comment by chriscool** [comment]
It looks like the commit has no "Signed-off-by: .* <.*>" line, only a "License: MIT" line (see https://github.com/ipfs/go-ipfs/pull/2529/commits/dc80a3dee519a4f0ab5920a8a0e268ec9726d7a5).

**Comment by thomas-gardner** [comment]
Ack---nothing gets past this cop. 

While you guys are here, have you tried this out yet?

File: misc/completion/ipfs-completion.bash [hunk 2]
```diff
@@ -1,421 +1,84 @@
...
-    _ipfs_comp "--all --help"
+	COMPREPLY=( $(compgen -W "$reply" -- "$cur") )
+	return 0
 }
 
-_ipfs_cat()
-{
-    _ipfs_help_only
+__ipfs_pin_list() {
+	ipfs refs local 2>/dev/null
```
Comment (whyrusleeping) [inline misc/completion/ipfs-completion.bash]: I would be careful about using this, its not a fast call (especially when you have gigabytes of content)

**Comment by whyrusleeping** [comment]
I havent tried this out yet since I don't use bash (fish shell ftw). If some bash users could give this a shot and report back, that would be very helpful

**Comment by thomas-gardner** [comment]
I looked at this recently and it's now completely broken. Will checkout the other PR.

**Pull request closed without merging**